### PR TITLE
[release/8.0] Ensure xUnit project is in .sln file & fix hyphen in name

### PR DESCRIPTION
**Pull request by DamianEdwards**
This is a manual backport of #3959 for GA that only includes the changes for the xUnit scenarios that was already approved in shiproom.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4023)

### CODE REVIEW

File: src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.sln [hunk 1]
```diff
@@ -8,7 +8,7 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "AspireStarterApplication.1.
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "AspireStarterApplication.1.ApiService", "AspireStarterApplication.1.ApiService\AspireStarterApplication.1.ApiService.csproj", "{9FEB877E-015D-4E20-AE63-06C596E242E4}"
 EndProject
-#if (CreateTestsProject)
```
Comment (timheuer) [inline src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.sln:11]: This is not a part of release/8.0 -- that still has CreateTestsProject

File: src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.Tests/WebTests.cs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 using System.Net;
 
-namespace AspireStarterApplication.1.Tests;
```
Comment (timheuer) [inline src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.Tests/WebTests.cs:3]: Also need to change in line 11 to add the `_1`

**Review by timheuer — CHANGES_REQUESTED** [review]

File: src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.sln (hunk 1 above)
Comment (DamianEdwards) [reply src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.sln:11]: bleurgh, of course

File: src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.Tests/WebTests.cs (hunk 2 above)
Comment (DamianEdwards) [reply src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.Tests/WebTests.cs:3]: Doesn't that line already have an underscore though?

Comment (DamianEdwards) [reply src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.Tests/WebTests.cs:3]: I see now your original PR had that change so I'll change it here too.

**Review by timheuer — APPROVED** [review]
lgtm now

**Review by joperezr — APPROVED** [review]

**Review by danmoseley — APPROVED** [review]
we don't need this part, because it's already fine for our xunit template?

<< #if (CreateTestsProject)
`>> #if (TestFramework != 'None')

**Comment by DamianEdwards** [comment]
@danmoseley the change to use the `TestFramework` symbol is only in main, not in release/8.0, as it was introduced to support the MSTest and NUnit variants.

**Pull request merged**